components/cdc/src/observer.rs

```diff
@@ -138,12 +160,31 @@ impl<E: KvEngine> CmdObserver<E> for CdcObserver {
 }
 
 impl RoleObserver for CdcObserver {
-    fn on_role_change(&self, ctx: &mut ObserverContext<'_>, role: StateRole) {
+    fn on_transfer_leader(&self, ctx: &mut ObserverContext<'_>, transferee: &Peer) {
+        self.maybe_add_leader_transferee(ctx.region().get_id(), transferee);
+    }
+
+    fn on_role_change(
+        &self,
+        ctx: &mut ObserverContext<'_>,
+        role: StateRole,
+        leader: Option<&Peer>,
+    ) {
         if role != StateRole::Leader {
             let region_id = ctx.region().get_id();
-            if let Some(observe_id) = self.is_subscribed(region_id) {
+            if let Some(ObserveState {
+                observe_id,
+                leader_transferee,
+            }) = self.is_subscribed(region_id)
+            {
                 // Unregister all downstreams.
-                let store_err = RaftStoreError::NotLeader(region_id, None);
+                let leader = if let Some(leader) = leader {
+                    Some(leader.to_owned())
+                } else {
+                    // Use leader transferee if leader is None.
+                    leader_transferee.clone()
```

Comment:
How about use https://github.com/tikv/raft-rs/blob/master/src/raft.rs#L202? It's maintained in raft-rs correctly. Otherwise we need also `on_transfer_leader_finished` to clear the new added field.